hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeysRenameRequest.java

```diff
@@ -0,0 +1,360 @@
+
+        try {
+          // check Acls to see if user has access to perform delete operation
+          // on old key and create operation on new key
+          checkKeyAcls(ozoneManager, volumeName, bucketName, fromKeyName,
+              IAccessAuthorizer.ACLType.DELETE, OzoneObj.ResourceType.KEY);
+          checkKeyAcls(ozoneManager, volumeName, bucketName, toKeyName,
+              IAccessAuthorizer.ACLType.CREATE, OzoneObj.ResourceType.KEY);
+        } catch (Exception ex) {
+          renameStatus = false;
+          unRenamedKeys.add(renameKeyArgs);
+          LOG.error("Acl check failed for fromKeyName {} toKeyName {}",
+              fromKeyName, toKeyName, ex);
+          continue;
+        }
+
+        // Check if toKey exists
+        String fromKey = omMetadataManager.getOzoneKey(volumeName, bucketName,
+            fromKeyName);
+        String toKey =
+            omMetadataManager.getOzoneKey(volumeName, bucketName, toKeyName);
+        OmKeyInfo toKeyValue = omMetadataManager.getKeyTable().get(toKey);
+
+        if (toKeyValue != null) {
+
+          // Check if this transaction is a replay of ratis logs.
+          if (isReplay(ozoneManager, toKeyValue, trxnLogIndex)) {
+
+            // Check if fromKey is still in the DB and created before this
+            // replay.
+            // For example, lets say we have the following sequence of
+            // transactions.
+            //   Trxn 1 : Create Key1
+            //   Trnx 2 : Rename Key1 to Key2 -> Deletes Key1 and Creates Key2
+            // Now if these transactions are replayed:
+            //   Replay Trxn 1 : Creates Key1 again it does not exist in DB
+            //   Replay Trxn 2 : Key2 is not created as it exists in DB and
+            //                   the request would be deemed a replay. But
+            //                   Key1 is still in the DB and needs to be
+            //                   deleted.
+            fromKeyValue = omMetadataManager.getKeyTable().get(fromKey);
+            if (fromKeyValue != null) {
+              // Check if this replay transaction was after the fromKey was
+              // created. If so, we have to delete the fromKey.
+              if (ozoneManager.isRatisEnabled() &&
+                  trxnLogIndex > fromKeyValue.getUpdateID()) {
+                acquiredLock =
+                    omMetadataManager.getLock().acquireWriteLock(BUCKET_LOCK,
+                        volumeName, bucketName);
+                // Add to cache. Only fromKey should be deleted. ToKey already
+                // exists in DB as this transaction is a replay.
+                Table<String, OmKeyInfo> keyTable = omMetadataManager
+                    .getKeyTable();
+                keyTable.addCacheEntry(new CacheKey<>(fromKey),
+                    new CacheValue<>(Optional.absent(), trxnLogIndex));
+                renameKeyInfoList.add(new OmRenameKeyInfo(
+                    null, fromKeyValue));
+              }
+            }
```

Comment:
Replay logic was removed in 90e8211bb39d7ecbbdd0a4b4df55e5a5aa64c932.  Can you please update the PR (also to resolve conflicts)?